components/infobox/wikis/starcraft2/infobox_person_player.lua

```diff
@@ -0,0 +1,556 @@
+		or Variables.varDefault('disable_SMW_storage', 'false') == 'true'
+		or mw.title.getCurrentTitle().nsText ~= ''
+	then
+		Variables.varDefine('disable_SMW_storage', 'true')
+		return false
+	end
+	return true
+end
+
+function StarCraft2Player.getAchievements(args)
+	local player = args.id or pagename
+	local achievements = Achievements({}, player)
+	if achievements == '' then
+		achievements = nil
+	else
+		hasAchievements = true
+	end
+	return achievements
+end
+
+--kick the default history display
+function StarCraft2Player.getHistory() return nil end
+
+function StarCraft2Player.addCustomContent(player, infobox, args)
+	local retired
+	--only display retired if it contains a year
+	if string.match(args.retired or '', '%d%d%d%d%') then
+		retired = args.retired
+	end
+	local allkills
+	if shouldStoreData then
+		allkills = StarCraft2Player._getAllkills()
+		if allkills then
+			allkills = _ALLKILLICON .. allkills
+		end
+	end
+	infobox:header('Achievements', (not hasAchievements) and allkills or nil)
+	infobox:cell('All-kills', allkills)
+	infobox:header('History', args.history)
+	infobox:centeredCell(args.history)
+	infobox:cell('Retired', retired)
+
+	return infobox
+end
+
+function StarCraft2Player._getAllkills()
+	if shouldStoreData then
+		local allkillsData = mw.ext.LiquipediaDB.lpdb('datapoint', {
+			conditions = '[[pagename::' .. pagename .. ']] AND [[type::allkills]]',
+			query = 'information',
+			limit = 1
+		})
+		if type(allkillsData[1]) == 'table' then
+			return allkillsData[1].information
+		end
+	end
+end
+
+function StarCraft2Player.getRole(_, args)
+	local role = args.role or args.occupation or 'player'
+	role = string.lower(role)
+	local ROLES = {
+		['admin'] = 'Admin', ['analyst'] = 'Analyst', ['coach'] = 'Coach',
+		['commentator'] = 'Commentator', ['caster'] = 'Commentator',
+		['expert'] = 'Analyst', ['host'] = 'Host', ['streamer'] = 'Streamer',
+		['interviewer'] = 'Interviewer', ['journalist'] = 'Journalist',
+		['manager'] = 'Manager', ['map maker'] = 'Map maker',
+		['observer'] = 'Observer', ['photographer'] = 'Photographer',
+		['tournament organizer'] = 'Organizer', ['organizer'] = 'Organizer',
+	}
+	local cleanOther = {
+		['blizzard'] = 'Blizzard', ['coach'] = 'Coach', ['staff'] = 'false',
+		['content producer'] = 'Content producer', ['streamer'] = 'false',
+	}
+	local category = ROLES[role]
+	local store = category or cleanOther[role] or 'Player'
+
+	return { title = 'Race', display = raceData.display, store = store, category = category or 'Player'}
+end
+
+function StarCraft2Player.calculateEarnings(_, args)
+	shouldStoreData = Player.shouldStoreData(args)
+
+	if shouldStoreData then
+		local earningsTotal
+		earningsTotal, earningsGlobal = StarCraft2Player._get_earnings_and_medals_data(pagename)
+		earningsTotal = math.floor( (earningsTotal or 0) * 100 + 0.5) / 100
+		return earningsTotal
+	end
+	return 0
+end
+
+function StarCraft2Player._get_earnings_and_medals_data(player)
```

Comment:
Function name doesn't follow guidelines

Reply:
yeah it is a temp copy paste from an old module (for testing)
it will get moved to an extra module later on